documentation/source/Segmentation.md

```diff
@@ -0,0 +1,152 @@
+# Image Segmentation
+
+SuperGradients allows users to train models for semantic segmentation tasks.
+The library includes pre-trained models, such as the Cityscapes PPLiteSeg model, and provides a simple interface for 
+loading custom datasets. 
+
+In the tutorial provided, we demonstrate how to fine-tune PPLiteSeg on a subset of the Supervisely dataset.
+You can run the following code in our [google collab](https://colab.research.google.com/drive/1d7cU0NsUj7jnOF1YSap_DH9r79G3-Cr4?usp=sharing#scrollTo=GqH4VGMroWec).
+
+## Load a dataset
+In this example we will work with supervisely-persons. If it's the first time you are using this dataset, or if you want to use another dataset please check out [dataset setup instructions](...)
+```py
+from super_gradients.training import dataloaders
+
+root_dir = '/path/to/supervisely_dataset_dir'
+
+train_loader = dataloaders.supervisely_persons_train(dataset_params={"root_dir": root_dir}, dataloader_params={})
+valid_loader = dataloaders.supervisely_persons_val(dataset_params={"root_dir": root_dir}, dataloader_params={})
+```
+
+
+### Visualization
+Let's visualize what we've got there.
+
+We have images and labels, with the default batch size of 256 for training.
+```py
+from PIL import Image
+from torchvision.utils import draw_segmentation_masks
+from torchvision.transforms import ToTensor, ToPILImage, Resize
+import numpy as np
+import torch
+
+def plot_seg_data(img_path: str, target_path: str):
+  image = (ToTensor()(Image.open(img_path).convert('RGB')) * 255).type(torch.uint8)
+  target = torch.from_numpy(np.array(Image.open(target_path))).bool()
+  image = draw_segmentation_masks(image, target, colors="red", alpha=0.4)
+  image = Resize(size=200)(image)
+  display(ToPILImage()(image))
+
+for i in range(4, 7):
+  img_path, target_path = train_loader.dataset.samples_targets_tuples_list[i]
+  plot_seg_data(img_path, target_path)
+```
+![segmentation_target.png](segmentation_target.png)
+
+
+## Load the model from modelzoo
+
+Create a PPLiteSeg nn.Module, with 1 class segmentation head classifier. For simplicity `use_aux_head` is set as `False`
+and extra Auxiliary heads aren't used for training.
+
+```py
+from super_gradients.training import models
+from super_gradients.common.object_names import Models
+
+model = models.get(
+    model_name=Models.PP_LITE_T_SEG75,      # You can use any model listed in the Models.<Name>
+    arch_params={"use_aux_heads": False},
+    num_classes=1,                          # Change this if you work on another dataset with more classes
+    pretrained_weights="cityscapes"         # Drop this line to train from scratch
+)
+```
+Notes:
```

Comment:
please add a comment here: model is a torch.nn.Module

Reply:
Changed in d1fb7168b1580600f5594a42819fa8246042c207